go-controller/pkg/ovn/egressip.go

```diff
@@ -1958,6 +2055,33 @@ type egressIPController struct {
 	egressIPNodeHealthCheckPort int
 }
 
+func (oc *Controller) addStandByEgressIPAssignment(podKey string, podStatus *podAssignmentState) error {
```

Comment:
this function is a little odd how it is called with a lock, then we unlock, call another function that locks/unlocks, then we lock again. Since this function is only called in one place, it might make more sense to just embed this logic there rather than create this new function. If you want to keep this function, then I think you need some function comments explaining this must be called with a lock and returns locked as well

Reply:
yeah I admit this seems a bit odd, there is another alternative which is to call `addPodEgressIPAssignments` directly from here. In that case it might be more cleaner in terms of locking logic though we might have to redo some bits from `addEgressIPAssignments`.